Sources/SwiftProtobuf/JSONEncodingVisitor.swift

```diff
@@ -374,7 +374,7 @@ internal struct JSONEncodingVisitor: Visitor {
   ) throws {
     try startField(for: fieldNumber)
     encoder.append(text: "{")
-    var mapVisitor = JSONMapEncodingVisitor(encoder: JSONEncoder(), options: options)
```

Comment:
Is this different than what was on `main`?

Reply:
`main` uses a new `JSONEncoder()` and this branch seems to require `self.encoder`